sycl/test/basic_tests/memory-consumption.cpp

```diff
@@ -0,0 +1,62 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// UNSUPPORTED: windows
+//
+//==-----memory-consumption.cpp - SYCL memory consumption basic test ------==//
```

Comment:
I think this test is valuable to check overall memory consumption. But I suggest adding more specific test on the change you are doing, such a verification could be done by setting SYCL_PI_TRACE and checking its output(see sycl/test/scheduler/ReleaseResourcesTest.cpp as an example).

Reply:
I will make a follow-up patch for this SYCL_PI_TRACE later.

Reply:
I'm OK if additional test will be added in the follow up patch.